### ISSUE TITLE
Threads: Context menu

### ISSUE THREAD

**Issue by daniellekirkwood** [vector-im/element-android#4688]
### Your use case

The context menu is large and can be overpowering. Do we need it this long every time?

**Comment by janogarcia** [comment]
@daniellekirkwood @ariskotsomitopoulos In the short term I'd suggest limiting the bottom sheet height to about 50% of the screen upon revealing it, to make it less overwhelming. Please check the **Visibility** section for the [Modal bottom sheet](https://material.io/components/sheets-bottom#modal-bottom-sheet) component in Material Design:

| Sheet height | Visibility and behavior |
| ------------- | ------------- |
| 50 to 100% of screen height  | Partially visibility at 50% of screen height. <br> On scroll or surface tap, reveal full sheet.  |

---

That's for the short term. In the medium term, I'd say the contextual actions are in desperate need of a review to bring some cross-platform cohesion (clean-up unnecessary actions, consistent ordering, wording, icons...).

**Comment by daniellekirkwood** [comment]
@janogarcia would the short-term solution leave us with a scroll in the bottom sheet?

**Comment by janogarcia** [comment]
@daniellekirkwood I'd say that depends on the device's viewport. But, yes, we should definitely observe the case where the viewport is actually shorter than the bottom sheet, which is not the case for the current mockups. Let me add that to Figma and then I'll follow up here.

**Comment by ariskotsomitopoulos** [comment]
@janogarcia @daniellekirkwood other than that, I believe we should remove some items from the list. The list should contain only **important** and **quick**  interactions. We should remove `Ignore user` while this should be when the user clicks on a user not on **every** message. We can also combine the share + permalink. We can also combine some similar items in the same line like **Reply | Quote**!

**Comment by daniellekirkwood** [comment]
Happy for you, @janogarcia, to work with @ariskotsomitopoulos and remove anything from the context menu that does not need to be there for Threads 👍

**Comment by janogarcia** [comment]
@ariskotsomitopoulos @daniellekirkwood I agree that we should basically refactor the available actions in there, but I'd recommend doing that as a separate effort, and in a cross-platform way. This is not affecting just threads, it's about the global contextual actions for messages, anywhere. I'd be worried of opening yet another can of worms (as such cross-platform changes may lead to multiple discussions) that could deviate our current focus on Threads.

That's what I actually meant by:

>  In the medium term, I'd say the contextual actions are in desperate need of a design review to bring some cross-platform cohesion (clean-up unnecessary actions, consistent ordering, wording, icons...).

In the meantime, what I would recommend is improving the bottom sheet component, so that it can fit a variable number of items, even on small viewports, but only displaying a limited set of them initially.

Also, adding some subtle signifiers (like the grabber on the top, or the rounded corners) will help in communicating the drag/scroll interactions. 

**Mockup**

→ [Figma: Bottom sheet Component](https://www.figma.com/file/T309ztx0sNyOOK6NKVLHsK/Threads?node-id=2854%3A288832)

**Snapshot**

![Threads – Figma](https://user-images.githubusercontent.com/60790/146223011-ca69fa2c-6dfe-4ec4-aa33-6f7937d9023c.png)

**Comment by daniellekirkwood** [comment]
I don't think we should consider taking this issue on for Classic Element. Instead we should design Element X knowing that we discussed these items here... @nadonomy Are you happy for me to close this issue - or should it be moved/recreated on an EX Design backlog somewhere?
